addons/sale/wizard/sale_make_invoice_advance.py

```diff
@@ -311,9 +311,19 @@ def _prepare_down_payment_lines_values(self, order):
         if self.advance_payment_method == 'percentage':
             percentage = self.amount / 100
         else:
-            percentage = self.fixed_amount / order.amount_total if order.amount_total else 1
-
-        order_lines = order.order_line.filtered(lambda l: not l.display_type and not l.is_downpayment)
+            invoice_lines = order.order_line.invoice_lines.filtered(
+                    lambda aml: aml.invoice_status == 'invoiced'
+                )
+            invoiced_total = sum(invoice_lines.price_total)
+            amount_total = order.amount_total - invoiced_total
+            percentage = 1 if order.currency_id.is_zero(amount_total) else self.fixed_amount / amount_total
```

Comment:
Line to long, either use a an if-else statement instead of an ternary expression, or shorten the `amount_total` variable name, e.g.
```suggestion
            amount = order.amount_total - invoiced_total
            percentage = 1 if order.currency_id.is_zero(amount) else self.fixed_amount / amount
```